testData/src/liquidityBootstrappingPool.ts

```diff
@@ -0,0 +1,169 @@
+import {
+    PublicClient,
+    createPublicClient,
+    http,
+    type Address,
+    type Chain,
+} from 'viem';
+import { CHAINS } from '@balancer/sdk';
+import { VAULT_V3, vaultExtensionAbi_V3 } from '@balancer/sdk';
+import { liquidityBootstrappingAbi } from './abi/liquidityBootstrapping';
+
+export type LBPoolImmutableData = {
+    tokens: string[];
+    decimalScalingFactors: bigint[];
+    startWeights: bigint[];
+    endWeights: bigint[];
+    startTime: bigint;
```

Comment:
A bunch of these are not being used anywhere which highlights that we're not taking into account the currentTime, etc. I was under the impression the implementation was going to accept a current time and calculate weights?

Reply:
Regarding calculating weights. In a past conversation we agreed that the desync between api providing weights and "true" onchain weights would be minimal enough that using them from the api would suffice. This is the reason why they are not calculated but rather simply provided in the constructor

Reply:
@johngrantuk I have added weight calculation and am utilizing these entries. Except decimalScaling factors, which however is also part of the WeightedPool data being generated.

Reply:
scalingFactors is part of the [BasePoolState](https://github.com/balancer/balancer-maths/blob/8d4b39401fed8ca5d1b035f79d18e93777797f73/typescript/src/vault/types.ts#L13) so will always be used by the Vault.
Can you change `decimalScalingFactors` to `scalingFactors` just to be consistent with the other pool implementations.

Reply:
Done, thanks for highlighting.